src/core/s3-defaults.ts

```diff
@@ -0,0 +1,16 @@
+import { RemovalPolicy } from 'aws-cdk-lib';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import { Bucket, BucketProps } from 'aws-cdk-lib/aws-s3';
+
+export function DefaultS3BucketProps(loggingBucket?: Bucket, lifecycleRules?: s3.LifecycleRule[]): s3.BucketProps {
+  return {
+    encryption: s3.BucketEncryption.S3_MANAGED,
+    versioned: false,
+    blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+    removalPolicy: RemovalPolicy.RETAIN,
+    enforceSSL: true,
+    eventBridgeEnabled: false,
+    ...((lifecycleRules !== undefined) && { lifecycleRules }),
```

Comment:
What does this syntax do exactly? Does it check if `lifecycleRules` is set, and if it is, it puts it in the props?

Reply:
That is my understanding indeed. If lifecycle rules is passed, add it to the object, otherwise ignore

Reply:
This is very cool, I've been missing that in `Python` a LOT